src/object/srv_obj_remote.c

```diff
@@ -509,8 +522,14 @@ ds_obj_cpd_dispatch(struct dtx_leader_handle *dlh, void *arg, int idx,
 		tgt_ep.ep_rank, tgt_ep.ep_tag, idx, DP_DTI(&dcsh->dcsh_xid));
 
 	rc = crt_req_send(req, shard_cpd_req_cb, remote_arg);
-	if (rc != 0)
+	if (rc != 0) {
+		if (unlikely(!sub->dss_comp)) {
+			do_shard_cpd_req_cb(req, remote_arg, rc);
+			crt_req_decref(req);
+		}
```

Comment:
I see this pattern is repeated in several places, it's better to make it into a wrapper function.

Reply:
Just look similar, but not the same, for different caller, they have different callback.